packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -5,14 +5,43 @@
 import 'basic.dart';
 import 'framework.dart';
 
-/// A place in the overlay that can contain a widget.
+/// A place in an [Overlay] that can contain a widget.
+///
+/// Overlay entires are inserted into an [Overlay] using the
+/// [OverlayState.insert] or [OverlayState.insertAll] functions. To find the
+/// closest enclosing overlay for a given [BuildContext], use the [Overlay.of]
+/// function.
+///
+/// An overlay entry can be in at most one overlay at a time. To remove an entry
+/// from its overlay, call the [remove] function on the overlay entry.
+///
+/// For example, [Draggable] uses an [OverlayEntry] to show the drag avatar that
+/// follows the user's finger across the screen after the drag begins. Using the
+/// overlay to display the drag avatar lets the avatar float over the other
```

Comment:
We generally follow the terminology from the web rather than from Android.  The class is called `_DragAvatar`, which is why I've used this term here.  I can't link to the class because its private but if someone looks at the draggable implementation they'll see it.